common/src/main/java/io/quarkus/ts/openshift/common/deploy/UsingQuarkusPluginDeploymentStrategy.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.ts.openshift.common.deploy;
+
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.quarkus.ts.openshift.common.Command;
+import io.quarkus.ts.openshift.common.injection.TestResource;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * If the test class is using the {@code UsingQuarkusPluginDeploymentStrategy} strategy, the test framework will use the
+ * Quarkus OpenShift extension to deploy the Quarkus Application by using the property `quarkus.kubernetes.deploy`.
+ *
+ */
+public class UsingQuarkusPluginDeploymentStrategy implements DeploymentStrategy {
+
+    private static final String MVN_COMMAND = "mvn";
+    private static final String PACKAGE_GOAL = "package";
+    private static final String QUARKUS_PLUGIN_DEPLOY = "-Dquarkus.kubernetes.deploy=true";
+    private static final String SKIP_TESTS = "-DskipTests=true";
+    private static final String SKIP_ITS = "-DskipITs=true";
+    private static final String QUARKUS_KUBERNETES_CLIENT_NAMESPACE = "quarkus.kubernetes-client.namespace";
+    private static final String QUARKUS_KUBERNETES_CLIENT_TRUST_CERTS = "quarkus.kubernetes-client.trust-certs";
+    private static final String QUARKUS_CONTAINER_IMAGE_GROUP = "quarkus.container-image.group";
+    private static final String QUARKUS_NATIVE_BUILDER_IMAGE = "quarkus.native.builder-image";
+    private static final String QUARKUS_NATIVE_CONTAINER_RUNTIME = "quarkus.native.container-runtime";
+    private static final String QUARKUS_PACKAGE_TYPE = "quarkus.package.type";
+
+    private static final String NATIVE = "native";
+    private static final String DOCKER = "docker";
+
+    @TestResource
+    private OpenShiftClient openShiftClient;
+
+    @Override
+    public void deploy() throws Exception {
+        String namespace = openShiftClient.getNamespace();
+
+        List<String> args = new ArrayList<>(
+                Arrays.asList(MVN_COMMAND, PACKAGE_GOAL, QUARKUS_PLUGIN_DEPLOY, SKIP_TESTS, SKIP_ITS));
```

Comment:
We should add `-B -V` here to avoid all the useless download progress logs and on the other hand to see what the Maven is.

Reply:
Good idea!

Reply:
Added.